### PR TITLE
feat(herbs2rest): using Herbarium to create REST endpoints

### PR DESCRIPTION
**Pull request by Anderson-Costa-Vx**
#22

Fixes #

## Proposed Changes

1. Using Herbarium to create REST endpoints

### Author/Contributor
- [ ] If documentation is needed for this change, has that been included in this pull request
- [ ] Remember to check if code coverage decrease, if so, please implement the necessary tests

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

File: src/herbs2rest.js
Comment (dalssoft) [inline src/herbs2rest.js:1]: recommended:
```js
module.exports = {
  generateRoutes: require('./generateRoutes'),
  generateControllers: require('./generateControllers') // former: generateRoutesWithHerbarium
}
```

this implies not only changing this file but also changing the `generateRoutesWithHerbarium` exports (including its own name)

File: src/generateRoutesWithHerbarium.js
Comment (dalssoft) [inline src/generateRoutesWithHerbarium.js:45]: should export only `generateControllers`

File: package.json
Comment (dalssoft) [inline package.json:63]: should be `devDependencies`

File: README.md
Comment (dalssoft) [inline README.md:79]: `As an alternative in a simpler way....`

This is not an alternative way, this is going to be the main way to use this library. The other ways should be "Advanced" and the doc should reflect that

File: test/generateRoutesWithHerbarium.test.js
Comment (dalssoft) [inline test/generateRoutesWithHerbarium.test.js:5]: do we need this dependencies for tests (express and supertest)? @Anderson-Costa-Vx @jhomarolo 
is it possible to stub it?

File: test/generateRoutesWithHerbarium.test.js
Comment (dalssoft) [inline test/generateRoutesWithHerbarium.test.js:4]: `require('@herbsjs/herbs')`

**Review by dalssoft — CHANGES_REQUESTED** [review]

File: test/generateRoutesWithHerbarium.test.js
Comment (jhomarolo) [reply test/generateRoutesWithHerbarium.test.js:5]: @dalssoft since we use `supertest` inside other test files to check de integrity and the status code of the endpoint, I don't see a big problem to keep this inside this PR.

I agree with you we should try to remove them, but for me, this is another issue to discuss

**Comment by jhomarolo** [comment]
@dalssoft I think we have a different problem here with aloe, a kind of circular reference issue maybe.

The CI is breaking because the aloe is requesting buchu to work, but the herbs2rest does not need it. When buchu was added to dev dependencies, worked.

I think this PR is good to go if you are done, but we need to discuss this. Maybe by removing the aloe from herbs core or removing the peerdependency from aloe, the issue will be resolved.

**Comment by jhomarolo** [comment]
@Anderson-Costa-Vx your PR was approved, but we won't merge until we figure out the aloe dependency problem, ok?

**Comment by Anderson-Costa-Vx** [comment]
> @Anderson-Costa-Vx your PR was approved, but we won't merge until we figure out the aloe dependency problem, ok?

Ok. Thanks.

**Comment by dalssoft** [comment]
> The CI is breaking because the aloe is requesting buchu to work, but the herbs2rest does not need it. 

I wasn't able to reproduce the problem. Do we still have a problem? How is it possible to reproduce?

Meanwhile, we could start the issue https://github.com/herbsjs/herbs-cli/issues/125, right?

**Comment by jhomarolo** [comment]
@dalssoft the problem is happening only inside the CI (there is should be a way to reproduce this) of Github and you need to remove buchu from dependencies

**Pull request merged**
